### PR TITLE
Add fetch step in the build pipeline

### PR DESCRIPTION
**Pull request by nikithauc**

### CODE REVIEW

File: .azure-pipelines/ci-build-production.yml [hunk 1]
```diff
@@ -16,7 +16,9 @@ pool:
 
 steps:
 
-- script: git checkout main
+- script: |
```
Comment (zengin) [inline .azure-pipelines/ci-build-production.yml:19]: any reason why we are not using the default `checkout` step?

**Review by zengin — CHANGES_REQUESTED** [review]

File: .azure-pipelines/ci-build-production.yml (hunk 1 above)
Comment (nikithauc) [reply .azure-pipelines/ci-build-production.yml:19]: The checkout step failed 

https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=104684&view=logs&j=6b902995-b73d-5f5c-66fd-a7f66c857d2c&t=7cbfe919-8ac5-5a5d-dd4b-f8ecdf2baae9

Added the fetch step based on 
https://stackoverflow.com/a/7334027/10760931

Comment (zengin) [reply .azure-pipelines/ci-build-production.yml:19]: I meant like this:

https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/blob/89258fcb74ae1f58d632d3d4632c2efeab3d1d0f/.azure-pipelines/generation-templates/checkout-metadata.yml#L1

Comment (nikithauc) [reply .azure-pipelines/ci-build-production.yml]: @zengin  Can you please check and let me know if that works?

**Review by zengin — APPROVED** [review]

**Pull request merged**